opentelemetry-exporter-geneva/geneva-uploader/src/payload_encoder/otlp_encoder.rs

```diff
@@ -567,12 +569,30 @@ impl OtlpEncoder {
                 }
                 FIELD_TRACE_ID => {
                     let hex_bytes = Self::encode_id_to_hex::<32>(&span.trace_id);
-                    let hex_str = std::str::from_utf8(&hex_bytes).unwrap();
+                    // Hex encoding produces valid ASCII, log error if conversion fails (data loss)
+                    let hex_str = std::str::from_utf8(&hex_bytes).unwrap_or_else(|e| {
```

Comment:
The comment looks odd. It says hex encoding always produces valid ASCII and then goes on to say log error if conversion would fail. The conversion would never fail if we were guaranteed to work on valid ASCII bytes.

We should either use `expect` or update the comment.

```rust
let hex_str = std::str::from_utf8(&hex_bytes).expect("hex encoding always produces valid UTF-8");
 ```